### PR TITLE
chore: remove Trace type from protos, replace with Hop

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #6549

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/transfer/forwarding_test.go [hunk 1]
```diff
@@ -80,8 +80,8 @@ func (s *TransferForwardingTestSuite) TestForwarding_WithLastChainBeingICS20v1_S
 
 	t.Run("packets are relayed from A to B to C", func(t *testing.T) {
 		chainCDenom := transfertypes.NewDenom(chainADenom,
-			transfertypes.NewTrace(channelBtoC.Counterparty.PortID, channelBtoC.Counterparty.ChannelID),
-			transfertypes.NewTrace(channelAtoB.Counterparty.PortID, channelAtoB.Counterparty.ChannelID),
+			transfertypes.NewHop(channelBtoC.Counterparty.PortID, channelBtoC.Counterparty.ChannelID),
```
Comment (crodriguezvega) [inline e2e/tests/transfer/forwarding_test.go:83]: `NewHop` just in time! :)

File: modules/apps/transfer/types/forwarding.go [hunk 2]
```diff
@@ -61,12 +63,18 @@ func (h Hop) Validate() error {
...
-		return errorsmod.Wrapf(err, "invalid source channel ID %s", h.ChannelId)
+		return errorsmod.Wrapf(err, "invalid hop source channel ID %s", h.ChannelId)
 	}
 
 	return nil
 }
 
+// String returns the Hop in the format:
+// <portID>/<channelID>
+func (h Hop) String() string {
```
Comment (crodriguezvega) [inline modules/apps/transfer/types/forwarding.go:74]: Nice.

**Review by crodriguezvega — APPROVED** [review]
Thank you, @DimitrisJim.

File: modules/apps/transfer/types/trace.go
Comment (damiannolan) [inline modules/apps/transfer/types/trace.go:1]: could optionally move the remaining funcs in this file to `denom.go` and delete `trace.go` maybe

File: proto/ibc/applications/transfer/v2/token.proto [hunk 3]
```diff
@@ -19,14 +20,5 @@ message Denom {
...
-}
-
-// Trace represents the portID and channelID the token arrived through.
-// When a token is sent to a new chain, the portID and channelID of the
-// destination chain are added to a token's trace.
-message Trace {
-  option (gogoproto.goproto_stringer) = false;
-  string port_id                      = 1;
-  string channel_id                   = 2;
+  repeated ibc.applications.transfer.v1.Hop trace = 3 [(gogoproto.nullable) = false];
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply proto/ibc/applications/transfer/v2/token.proto:23]: I think it makes sense 👍🏻

**Review by damiannolan — APPROVED** [review]
Nice, very thorough! 🙏🏻

**Pull request merged**